### ISSUE TITLE
The "logger" option is ignored

### ISSUE THREAD

**Issue by pjbatista** [rogierschouten/gulp-typedoc#21]
If you pass either a custom logger or the typedoc-accepted "none" value, gulp-doc will ignore it, create its own logger, and render the app version to the output.

That happens because it doesn't check for an options value before assigning it a new value:

```javascript
// index.js (line 39)
// reduce console logging
options.logger = function(message, level, newline) {
    if (level === 3) {
        gutil.log(gutil.colors.red(message));
    }
}
```

I have a fix proposal at my fork: https://github.com/pjbatista/gulp-typedoc. Let me know what you think, Rogier.

**Comment by rogierschouten** [comment]
Hi @pjbatista thank you.  I think that looks pretty good already. 

The only bit I'm unsure about is the part where you check for the presence of app.logger.hasErrors.   From the typedoc code, I cannot find any place where app.logger is set to options.logger (rather app.loggerType).  Therefore, hasErrors will always be present (otherwise the plugin would already have been in trouble because it sets a plain function as options.logger).

I think the best way forward would be for you submit a PR  (excluding the package.json change, and maybe the hasErrors check if what I say above is true)  and I'll be more than happy to test and integrate it. Agree?

Thanks again, well noticed.

**Comment by pjbatista** [comment]
You are correct! I got it mixed up with the logger property from the options object, which is of variable type. The logger property from the application will always be an instance of the Logger class (https://github.com/TypeStrong/typedoc/blob/master/src/lib/application.ts, line 54) thus will always contain hasErrors.

I'm changing it and submitting a PR.

Thanks, man!

**Comment by rogierschouten** [comment]
Fix published as v2.1.0  Thank you Pedro!
